build/test/ProjectsToTest.props

```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project DefaultTargets="Layout" ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+      <ItemGroup>
+            <ProjectsToTest Include="ArgumentForwardingTests"/>
+            <ProjectsToTest Include="crossgen.Tests"/>
+            <ProjectsToTest Include="EndToEnd"/>
+            <ProjectsToTest Include="dotnet.Tests"/>
+            <ProjectsToTest Include="dotnet-build.Tests"/>
+            <ProjectsToTest Include="dotnet-build3.Tests"/>
+            <ProjectsToTest Include="dotnet-compile.Tests"/>
+            <ProjectsToTest Include="dotnet-compile.UnitTests"/>
+            <!-- TODO: https://github.com/dotnet/cli/issues/3558 /><ProjectsToTest Include="/ "dotnet-compile-fsc.Tests" />-->
+            <ProjectsToTest Include="dotnet-new.Tests"/>
+            <ProjectsToTest Include="dotnet-pack.Tests"/>
+            <ProjectsToTest Include="dotnet-publish.Tests"/>
+            <ProjectsToTest Include="dotnet-resgen.Tests"/>
+            <ProjectsToTest Include="dotnet-run.Tests"/>
+            <ProjectsToTest Include="dotnet-run.UnitTests"/>
+            <ProjectsToTest Include="dotnet-test.Tests"/>
+            <ProjectsToTest Include="dotnet-test.UnitTests"/>
+            <!-- TODO: https://github.com/dotnet/cli/issues/321 /><ProjectsToTest Include="/"Kestrel.Tests" />-->
+            <ProjectsToTest Include="Microsoft.DotNet.Cli.Utils.Tests"/>
+            <ProjectsToTest Include="Microsoft.DotNet.Compiler.Common.Tests"/>
+            <ProjectsToTest Include="Microsoft.DotNet.ProjectModel.Tests"/>
+            <ProjectsToTest Include="Microsoft.DotNet.ProjectModel.Loader.Tests"/>
+            <ProjectsToTest Include="Microsoft.Extensions.DependencyModel.Tests"/>
+            <ProjectsToTest Include="Microsoft.DotNet.Configurer.UnitTests"/>
+            <ProjectsToTest Include="Performance"/>
+            <ProjectsToTest Condition="'$(DesktopAvailable)' == 'True'" Include="binding-redirects.Tests"/>
```

Comment:
Since $(DesktopAvailable) is defined inside of the "Init" target, the property value is not available at this point.  (MSBuild evaluates all static properties and items first, before it executes any targets.

To fix this, I think $(DesktopAvailable) should be set in build.proj in a static property (i.e. not inside a target).  You can use the `Condition=" '$(OS)' == 'Windows_NT' "` to check for windows.